app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddSidewalkSurface.kt

```diff
@@ -0,0 +1,125 @@
+                sidewalk ~ both|left|right or
+                sidewalk:both = yes or
+                (sidewalk:left = yes and sidewalk:right ~ yes|no|separate) or
+                (sidewalk:right = yes and sidewalk:left ~ yes|no|separate)
+            )
+            and (
+                ${SIDEWALK_SURFACE_KEYS.joinToString(" and ") {"!$it"}}
+                or sidewalk:surface older today -8 years
+            )
+    """
+
+    override val changesetComment = "Add surface of sidewalks"
+    override val wikiLink = "Key:sidewalk"
+    override val icon = R.drawable.ic_quest_sidewalk_surface
+    override val isSplitWayEnabled = true
+    override val questTypeAchievements = listOf(PEDESTRIAN, WHEELCHAIR)
+    override val defaultDisabledMessage = R.string.default_disabled_msg_difficult_and_time_consuming
+
+    override fun getTitle(tags: Map<String, String>) : Int =
+        R.string.quest_sidewalk_surface_title
+
+    override fun createForm() = AddSidewalkSurfaceForm()
+
+    override fun applyAnswerTo(answer: SidewalkSurfaceAnswer, tags: Tags, timestampEdited: Long) {
+        val leftChanged = answer.left?.let { sideSurfaceChanged(it, Side.LEFT, tags) }
+        val rightChanged = answer.right?.let { sideSurfaceChanged(it,  Side.RIGHT, tags) }
+
+        if (leftChanged == true) {
+            deleteSmoothnessKeys(Side.LEFT, tags)
+            deleteSmoothnessKeys(Side.BOTH, tags)
+        }
+        if (rightChanged == true) {
+            deleteSmoothnessKeys(Side.RIGHT, tags)
+            deleteSmoothnessKeys(Side.BOTH, tags)
+        }
+
+        if (answer.left == answer.right) {
+            answer.left?.let { applySidewalkSurfaceAnswerTo(it, Side.BOTH, tags) }
+            deleteSidewalkSurfaceAnswerIfExists(Side.LEFT, tags)
+            deleteSidewalkSurfaceAnswerIfExists(Side.RIGHT, tags)
+        } else {
+            answer.left?.let { applySidewalkSurfaceAnswerTo(it, Side.LEFT, tags) }
+            answer.right?.let { applySidewalkSurfaceAnswerTo(it, Side.RIGHT, tags) }
+            deleteSidewalkSurfaceAnswerIfExists(Side.BOTH, tags)
+        }
+        deleteSidewalkSurfaceAnswerIfExists(null, tags)
+
+        // only set the check date if nothing was changed or if check date was already set
+        if (!tags.hasChanges || tags.hasCheckDateForKey("sidewalk:surface")) {
+            tags.updateCheckDateForKey("sidewalk:surface")
+        }
+    }
+
+    private enum class Side(val value: String) {
+        LEFT("left"), RIGHT("right"), BOTH("both")
+    }
+
+    private fun sideSurfaceChanged(surface: SurfaceAnswer, side: Side, tags: Tags): Boolean {
+        val previousSideValue = tags["sidewalk:${side.value}:surface"]
+        val previousBothOsmValue = tags["sidewalk:both:surface"]
+        val osmValue = surface.value.osmValue
+
+        return if (previousSideValue != null && previousSideValue != osmValue) {
+            true
+        } else previousBothOsmValue != null && previousBothOsmValue != osmValue
```

Comment:
```suggestion
        val previousSideOsmValue = tags["sidewalk:${side.value}:surface"]
        val previousBothOsmValue = tags["sidewalk:both:surface"]
        val osmValue = surface.value.osmValue

        return previousSideOsmValue != null && previousSideOsmValue != osmValue
            || previousBothOsmValue != null && previousBothOsmValue != osmValue
```
consistent naming + simplify return statement